code/script.js

```diff
@@ -28,26 +40,223 @@ const showMessage = (message, sender) => {
+  } else {
+    showMessage(`Nice to meet you, ${userName}. Are you ready to solve some math problems?`, "bot")
+    //Buttons for answering "yes" or "no"
+    inputWrapper.innerHTML = `
+      <button id="yes" class="chat-btn" type="submit">Yes</button>
+      <button id="no" class="chat-btn" type="submit">No</button>
+    `
+    document.getElementById("yes").addEventListener("click", handleYes) 
+    document.getElementById("no").addEventListener("click", handleNo)
+  }
+}
+//If you answer "no"
+const handleNo = () => {
+  showMessage(`No`, "user")
+  showMessage(`Sorry to hear that, ${userName}. You're welcome back anytime!`, "bot")
+  inputWrapper.innerHTML = ``
+  setTimeout(() => clearWindow(), 4000)
+}
+//If you answer "yes"
+const handleYes = () => {
+  showMessage(`Yes`, "user")
+  setTimeout(() => askMathType(), 1000)
+}
+//Ask what type of question you want
+const askMathType = () => {
+  showMessage(`What type of math problem do you want to solve?`, "bot")
+  inputWrapper.innerHTML = `
+    <button id="add" type="submit" class="choice-btn">Addition</button>
+    <button id="subtract" type="submit" class="choice-btn">Subtraction</button>
+    <button id="multiply" type="submit" class="choice-btn">Multiplication</button>
+    <button id="divide" type="submit" class="choice-btn">Division</button>
+  `
+  document.getElementById("add").addEventListener("click", handleAddition)  
+  document.getElementById("subtract").addEventListener("click", handleSubtraction)
+  document.getElementById("multiply").addEventListener("click", handleMultiplication)  
+  document.getElementById("divide").addEventListener("click", handleDivision)   
+}
+//If you choose addition
+const handleAddition = () => {
+  mathType = "addition"
+  showMessage (`Addition`, "user")
+  setTimeout(() => askAdditionQuestion(), 1000)
+}
+//If you choose subtraction
+const handleSubtraction = () => {
+  mathType = "subtraction"
+  showMessage (`Subtraction`, "user")
+  setTimeout(() => askSubtractionQuestion(), 1000)
+}
+//If you choose multiplication
+const handleMultiplication = () => {
+  mathType = "multiplication"
+  showMessage(`Multiplication`, "user")
+  setTimeout(() => askMultiplicationQuestion(), 1000)
+}
+//If you choose division
+const handleDivision = () => {
+  mathType = "division"
+  showMessage(`Division`, "user")
+  setTimeout(() => askDivisionQuestion(), 1000)
+}
+//Prints a mathproblem with addition
+const askAdditionQuestion = () => {
+  a = Math.floor (Math.random() * 100) + 1
+  b = Math.floor(Math.random() * 100) + 1
+  showMessage (`What is ${a}+${b}?`, "bot")
+  inputWrapper.innerHTML = `
+    <form id="answerForm"> 
+      <label>Answer</label>
+      <input id="answer" type="text" />  
+      <button id="button" class="send-btn" type="submit"> 
+          Send
+      </button>
+    </form>
+  `
+  document.getElementById("answerForm").addEventListener("submit", handleAnswer)
+}
```

Comment:
Some code here seems to be reused in all the askMathtypeQuestion functions. Could you think of a way to refactor it to keep it more DRY?